File: server/src/main/java/org/paratranz/bot/bot/MiraiBotConfig.java
package org.paratranz.bot.bot;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import net.mamoe.mirai.Bot;
import net.mamoe.mirai.BotFactory;
import net.mamoe.mirai.utils.BotConfiguration;
import net.mamoe.mirai.utils.LoggerAdapters;
import org.paratranz.bot.api.ParatranzApi;
import org.paratranz.bot.bot.core.CommandProcessorBeanPostProcessor;
import org.paratranz.bot.bot.schedules.ApplicationSchedule;
import org.paratranz.bot.properties.ExternalProperties;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.boot.context.event.ApplicationReadyEvent;
import org.springframework.context.ApplicationListener;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Slf4j
@Configuration
@RequiredArgsConstructor
@ConditionalOnProperty(name = "system.module-control.mirai-active", havingValue = "true", matchIfMissing = true)
public class MiraiBotConfig implements ApplicationListener<ApplicationReadyEvent> {

    @Bean
    public ParatranzApi paratranzClient() {
        return new ParatranzApi("5828fb55756dbf6ebb4f76937c16e530");
    }

    @Bean
    public Bot miraiBot(
            ExternalProperties externalProperties
    ) {
        Bot bot = BotFactory.INSTANCE.newBot(
                externalProperties.getQq(),
                externalProperties.getPassword(),
                (config) -> {
                    config.setAutoReconnectOnForceOffline(true);
                    config.setProtocol(BotConfiguration.MiraiProtocol.ANDROID_PHONE);
                    config.setShowingVerboseEventLog(false);
                    config.setLoginCacheEnabled(true);
                    config.setReconnectionRetryTimes(10);
                    //覆盖Mirai内置的日志系统
                    config.setBotLoggerSupplier(it -> LoggerAdapters.asMiraiLogger(log));
                });
        //群聊事件处理
        bot.getEventChannel().registerListenerHost(new MiraiEventListener(bot.getCoroutineContext()));
        return bot;
    }

    @Bean
    public ApplicationSchedule applicationSchedule(
            ParatranzApi paraTranzApi,
            ExternalProperties externalProperties,
            Bot bot
    ) {
        return new ApplicationSchedule(paraTranzApi, externalProperties, bot);
    }

    @Bean
    public CommandProcessorBeanPostProcessor commandProcessorHandler() {
        return new CommandProcessorBeanPostProcessor();
    }

    @Override
    public void onApplicationEvent(ApplicationReadyEvent event) {
        event.getApplicationContext()
                .getBean(Bot.class)
                .login();
    }
}

File: server/src/main/java/org/paratranz/bot/bot/MiraiBotConfig.kt
package org.paratranz.bot.bot

import kotlinx.coroutines.runBlocking
import net.mamoe.mirai.Bot
import net.mamoe.mirai.BotFactory
import net.mamoe.mirai.utils.BotConfiguration.MiraiProtocol
import org.paratranz.bot.api.ParatranzApi
import org.paratranz.bot.bot.core.CommandProcessorBeanPostProcessor
import org.paratranz.bot.properties.ExternalProperties
import org.slf4j.LoggerFactory
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.boot.context.event.ApplicationReadyEvent
import org.springframework.context.ApplicationListener
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
@ConditionalOnProperty(name = ["system.module-control.mirai-active"], havingValue = "true", matchIfMissing = true)
open class MiraiBotConfig : ApplicationListener<ApplicationReadyEvent> {

    private val log = LoggerFactory.getLogger(this::class.java)

    @Bean
    open fun paraApi(): ParatranzApi {
        return ParatranzApi("sadddddddd")
    }

    @Bean
    open fun miraiBot(
        externalProperties: ExternalProperties
    ): Bot {
        val bot = BotFactory.newBot(externalProperties.qq, externalProperties.password) {
            this.autoReconnectOnForceOffline = true
            this.reconnectionRetryTimes = 30
            this.protocol = MiraiProtocol.ANDROID_PHONE
            this.isShowingVerboseEventLog = true
            this.loginCacheEnabled = true
        }
        //群聊事件处理
        bot.eventChannel.registerListenerHost(MiraiEventListener(bot.coroutineContext))
        return bot
    }

    @Bean
    open fun commandProcessorHandler(): CommandProcessorBeanPostProcessor {
        return CommandProcessorBeanPostProcessor()
    }

    /**
     * 开始执行登录
     */
    override fun onApplicationEvent(event: ApplicationReadyEvent) {
        runBlocking {
            event.applicationContext.getBean(Bot::class.java).login()
        }
    }
}
